src/VisualStudio/IntegrationTest/TestSetup/AsyncCompletionTracker.cs

```diff
@@ -48,7 +50,21 @@ private void HandleAsyncCompletionTriggered(object sender, CompletionTriggeredEv
 
             e.CompletionSession.Dismissed += ReleaseToken;
             e.CompletionSession.ItemCommitted += ReleaseToken;
-            e.CompletionSession.ItemsUpdated += ReleaseToken;
+
+            _ = Task.Run(async () =>
+                {
+                    // AsyncCompletion might fires multiple ItemsUpdated events perf keystroke typed, which means
+                    // we could see the first ItemsUpdated event even though items don't change (but computation finished).
+                    // If test attempts to assert state after seeing first event it would cause flakiness. 
+                    // Use SelectedItemProvider from the UI thread to wait for all pending work to be completed.
+
+#pragma warning disable RS0030 // Do not used banned APIs
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+#pragma warning restore RS0030 // Do not used banned APIs
```

Comment:
Is the banned-ness here just that we should be importing our IThreadingContext or something similar?